source/grid/grid_reordering.cc

```diff
@@ -29,585 +29,710 @@
-              while (side_hop(qqnum,lsn))
-                {
-                  // switch this face
-                  lsn = (lsn+2)%4;
-                  if (!is_oriented_side(qqnum,lsn))
-                    orient_side(qqnum,lsn);
-                  else
-                    //We've found a
-                    //cycle.. and
-                    //oriented all
-                    //quads in it.
-                    break;
-                }
-            }
-        }
+      for (unsigned int c=0; c<cells.size(); ++c)
+        for (unsigned int l=0; l<GeometryInfo<dim>::lines_per_cell; ++l)
+          if (edges[cells[c].edge_indices[l]].orientation_status == Edge<dim>::not_oriented)
+            return c;
+
+      return numbers::invalid_unsigned_int;
     }
 
 
+    /**
+     * Given a set of cells and edges, orient all edges that are
+     * (globall) parallel to the one identified by the @p cell and
+     * within it the one with index @p local_edge.
+     */
     void
-    GridReordering::orient_side(const unsigned int quadnum,
-                                const unsigned int localsidenum)
+    orient_one_set_of_parallel_edges (const std::vector<Cell<2> > &cells,
+                                      std::vector<Edge<2> >       &edges,
+                                      const unsigned int           cell,
+                                      const unsigned int           local_edge)
     {
-      MQuad &quad = mquads[quadnum];
-      int op_side_l = (localsidenum+2)%4;
-      MSide &side = sides[mquads[quadnum].side[localsidenum]];
-      const MSide &op_side = sides[mquads[quadnum].side[op_side_l]];
-
-      //is the opposite side oriented?
-      if (op_side.Oriented)
-        {
-          //YES - Make the orientations match
-          //Is op side in default orientation?
-          if (op_side.v0 == quad.v[ConnectGlobals::DefaultOrientation[op_side_l][0]])
-            {
-              //YES
-              side.v0 = quad.v[ConnectGlobals::DefaultOrientation[localsidenum][0]];
-              side.v1 = quad.v[ConnectGlobals::DefaultOrientation[localsidenum][1]];
-            }
-          else
-            {
-              //NO, its reversed
-              side.v0 = quad.v[ConnectGlobals::DefaultOrientation[localsidenum][1]];
-              side.v1 = quad.v[ConnectGlobals::DefaultOrientation[localsidenum][0]];
-            }
-        }
+      const unsigned int dim = 2;
+
+      // choose the direction of the first edge. we have free choice
+      // here and could simply choose "forward" if that's what pleases
+      // us. however, for backward compatibility with the previous
+      // implementation used till 2016, let us just choose the
+      // direction so that it matches what we have in the given cell.
+      //
+      // in fact, in what can only be assumed to be a bug in the
+      // original implementation, after orienting all edges, the code
+      // that rotates the cells so that they match edge orientations
```

Comment:
I don't understand the point. If it is a bug, why do you still keep this behavior? Or if you fixed the bug, there is no reason to talk about it here anymore, right?